app-android/src/main/java/com/ustadmobile/port/android/view/binding/TextViewBindings.kt

```diff
@@ -204,3 +206,66 @@ fun TextView.setHtmlText(htmlText: String?) {
 fun TextView.setFileSize(fileSize: Long) {
     text = UMFileUtil.formatFileSize(fileSize)
 }
+
+@BindingAdapter(value=["clazzMemberWithClazzWorkAndProgress"])
+fun TextView.setClazzWorkMarking(clazzMemberWithClazzWorkAndProgress: ClazzMemberWithClazzWorkProgress){
+    val marked = context.getString(R.string.marked_cap)
+    val completed = context.getString(R.string.completed)
+    val submitted = context.getString(R.string.submitted_cap)
+    val notSubmitted = context.getString(R.string.not_submitted_cap)
+    val ofContent = context.getString(R.string.of_content)
+
+    val clazzWorkSubmission = clazzMemberWithClazzWorkAndProgress.mClazzWorkSubmission
+    var line2Bit1 = ""
+    var line2Bit2 = ""
+    if(clazzWorkSubmission == null){
+        line2Bit1 = notSubmitted
+    }else{
+        line2Bit1 = if(clazzWorkSubmission.clazzWorkSubmissionDateTimeFinished > 0){
+            if(clazzWorkSubmission.clazzWorkSubmissionDateTimeMarked > 0){
+                marked
+            }else{
+                submitted
+            }
+        }else{
+            notSubmitted
+        }
+    }
+
+    if(clazzMemberWithClazzWorkAndProgress.clazzWorkHasContent &&
+            clazzMemberWithClazzWorkAndProgress.mProgress >= 0){
+        line2Bit2 = " - " + completed + " " + clazzMemberWithClazzWorkAndProgress.mProgress.toInt() +
```

Comment:
I would recommend creating an extension function to get the status of a ClazzWorkSubmission: e.g.

```
fun ClazzWorkSubmission?.statusString(context: Context) = when {
    this == null -> context.getString(R.string.notSubmitted)
    this.clazzWorkSubmissionDateTimeMarked > 0 -> context.getString(R.string.marked)
    this.clazzWorkSubmissionDateTimeFinished > 0 -> context.getString(R.string.submitted)
    else -> context.getString(R.string.notSubmitted)

}
```

I would also use string templates. I think once you have that function - this whole binding adapter could become:
```
fun TextView.setClazzWorkMarking(clazzMemberWithClazzWorkAndProgress: ClazzMemberWithClazzWorkProgress){
   var line = "${clazzMemberWithClazzWorkAndProgress.mClazzWorkSubmission.statusString(context)}"
   if(clazzMemberWithClazzWorkAndProgress.clazzWorkHasContent && clazzMemberWithClazzWorkAndProgress.mProgress >= 0) {
        line += " ${context.getString(R.string.completed)} ${clazzMemberWithClazzWorkAndProgress.mProgress.toInt()}% ${context.getString(R.string.of_content)}"
    }
    text = line
}
```